packages/cli-repl/src/cli-repl.ts

```diff
@@ -155,15 +155,25 @@ class CliRepl {
     await this.verifyNodeVersion();
 
     if (!this.cliOptions.nodb) {
+      let cs;
+      if (driverUri !== '') {
```

Comment:
I don’t think this condition is possible to reach – this is the return value of `generateUri()` in `run.ts`, which only returns an empty string if `nodb` is set:

https://github.com/mongodb-js/mongosh/blob/e6e75b455a65fad7b78c42571d170bc5cc1891ef/packages/service-provider-core/src/uri-generator.ts#L158-L171

So you can drop the conditional and make `cs` const (and make `driverUri = cs.href` the last line in this block, no need to duplicate the line anymore) :)

Reply:
Yeah, it will look much better! I didn't like to do `driverUri = cs.href` several times :) but I didn't know about the fact that `driverUri` is always a string and just moved this check to the beginning of the function: https://github.com/mongodb-js/mongosh/pull/1190/files#diff-d998d2f6015503912abc1073f3bdb46da4c12651a7ba3d2e6170711edb19b45aL160

I will get rid of it then! Thanks for sharing the place in code when the empty string is being returned!